### PR TITLE
rbd: implement rbdVolume.resize() with go-ceph

### PR DESCRIPTION
**Pull request by agarwal-mudit**
Replaced command execution with go-ceph Resize() function.

Signed-off-by: Mudit Agarwal <muagarwa@redhat.com>

Fixes: #1269

### CODE REVIEW

File: internal/rbd/rbd_util.go [hunk 1]
```diff
@@ -1043,18 +1043,16 @@ func cleanupRBDImageMetadataStash(path string) error {
...
+func (rv *rbdVolume) resize(newSize int64) error {
+	image, err := rv.open()
 	if err != nil {
-		return fmt.Errorf("failed to resize rbd image (%w), command output: %s", err, stderr)
+		return err
 	}
+	defer image.Close()
 
-	return nil
+	volSzMiB := util.RoundOffVolSize(newSize)
```
Comment (nixpanic) [inline internal/rbd/rbd_util.go:1054]: newSize is in MiB, but `image.Resize()` takes bytes. The e2e test shows the resize shrunk the volume:
```
Jul 31 03:06:46.241: INFO: At 2020-07-31 02:56:46 +0000 UTC - event for rbd-pvc: {external-resizer rbd.csi.ceph.com } FileSystemResizeRequired: Require file system resize of volume on node
Jul 31 03:06:46.241: INFO: At 2020-07-31 02:56:46 +0000 UTC - event for rbd-pvc: {volume_expand } ExternalExpanding: Ignoring the PVC: didn't find a plugin capable of expanding the volume; waiting for an external controller to process this PVC.
Jul 31 03:06:46.241: INFO: At 2020-07-31 02:56:46 +0000 UTC - event for rbd-pvc: {external-resizer rbd.csi.ceph.com } Resizing: External resizer is resizing volume pvc-e3e15379-56e0-43c3-9cc7-ad2e6e4e330f
Jul 31 03:06:46.241: INFO: At 2020-07-31 02:57:55 +0000 UTC - event for csi-rbd-demo-pod: {kubelet travis-job-8d071cfa-ab4e-4ae4-94a3-be6e13145e8e} VolumeResizeFailed: NodeExpandVolume.NodeExpandVolume failed for volume "pvc-e3e15379-56e0-43c3-9cc7-ad2e6e4e330f" : MountVolume.NodeExpandVolume failed : Expander.NodeExpand failed to expand the volume : rpc error: code = Unknown desc = rbd: resize failed on path /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-e3e15379-56e0-43c3-9cc7-ad2e6e4e330f/globalmount, error: resize of device /dev/rbd0 failed: exit status 1. resize2fs output: resize2fs 1.45.4 (23-Sep-2019)
resize2fs: New size smaller than minimum (12980)
```

Comment (agarwal-mudit) [reply internal/rbd/rbd_util.go:1054]: Strangely, it passed for me during normal resize operation. Will change it.

File: internal/rbd/controllerserver.go [hunk 2]
```diff
@@ -1090,13 +1090,12 @@ func (cs *ControllerServer) ControllerExpandVolume(ctx context.Context, req *csi
...
 	if rbdVol.VolSize < volSize {
 		util.DebugLog(ctx, "rbd volume %s size is %v,resizing to %v", rbdVol, rbdVol.VolSize, volSize)
-		rbdVol.VolSize = volSize
-		nodeExpansion = true
 		err = rbdVol.resize(volSize)
 		if err != nil {
 			klog.Errorf(util.Log(ctx, "failed to resize rbd image: %s with error: %v"), rbdVol, err)
 			return nil, status.Error(codes.Internal, err.Error())
 		}
+		nodeExpansion = true
```
[the remark this answers is not in the document]
Comment (agarwal-mudit) [reply internal/rbd/controllerserver.go:1098]: I have squashed the commits now.

**Comment by Madhu-1** [comment]
@MergifyIO rebase

**Pull request merged**